tcms/core/ajax.py

```diff
@@ -233,29 +233,28 @@ def add(self):
                     for o in self.obj:
                         o.add_tag(tag)
                 except IntegrityError:
-                    return "Error when adding %s " \
+                    return False, "Error when adding %s " \
```

Comment:
For all places where you modify the return type and return a tuple instead of string there needs to be tests. The fact that nothing fails means either there aren't any tests for these conditions or the tests are not written very well so they don't fail when you introduce the changes.